src/preferences/preferences.cpp

```diff
@@ -1339,145 +1339,150 @@ void Preferences::disableRecursiveDownload(bool disable) {
+
+    if (res == ERROR_SUCCESS) {
+        cMaxSubKeyLen++; // For null character
+        LPWSTR lpName = new WCHAR[cMaxSubKeyLen];
+        DWORD cName;
+
+        for (DWORD i = 0; i < cSubKeys; ++i) {
+            cName = cMaxSubKeyLen;
+            res = ::RegEnumKeyExW(handle, 0, lpName, &cName, NULL, NULL, NULL, NULL);
+            if (res == ERROR_SUCCESS)
+                keys.push_back(QString::fromWCharArray(lpName));
+        }
 
-QString getRegValue(const HKEY &handle, const QString &name = QString()) {
-  QString end_result;
-  DWORD type = 0;
-  DWORD size = 0;
-  DWORD array_size = 0;
+        delete[] lpName;
+    }
 
-  LPTSTR value_name = NULL;
-  if (!name.isEmpty()) {
-    value_name = new TCHAR[name.size()+1];
-    name.toWCharArray(value_name);
-    value_name[name.size()] = '\0';
-  }
+    return keys;
+}
 
-  // Discover the size of the value
-  ::RegQueryValueEx(handle, value_name, NULL, &type, NULL, &size);
-  array_size = size / sizeof(TCHAR);
-  if (size % sizeof(TCHAR))
-    array_size++;
-  array_size++; //For null character
-  LPTSTR value = new TCHAR[array_size];
-
-  long res = ::RegQueryValueEx(handle, value_name, NULL, &type, (LPBYTE)value, &size);
-  if (res == ERROR_SUCCESS) {
-    value[array_size] = '\0';
-    end_result = QString::fromWCharArray(value);
-  }
+QString getRegValue(HKEY handle, const QString &name = QString())
+{
+    QString result;
+
+    DWORD type = 0;
+    DWORD cbData = 0;
+    LPWSTR lpValueName = NULL;
+    if (!name.isEmpty()) {
+        lpValueName = new WCHAR[name.size() + 1];
+        name.toWCharArray(lpValueName);
+        lpValueName[name.size()] = 0;
+    }
 
-  if (value_name)
-    delete[] value_name;
-  if (value)
-    delete[] value;
+    // Discover the size of the value
+    ::RegQueryValueExW(handle, lpValueName, NULL, &type, NULL, &cbData);
+    DWORD cBuffer = (cbData / sizeof(WCHAR)) + 1;
+    LPWSTR lpData = new WCHAR[cBuffer];
+    LONG res = ::RegQueryValueExW(handle, lpValueName, NULL, &type, (LPBYTE)lpData, &cbData);
+    if (lpValueName)
+        delete[] lpValueName;
+
+    if (res == ERROR_SUCCESS) {
+        lpData[cBuffer] = 0;
+        result = QString::fromWCharArray(lpData);
+    }
+    delete[] lpData;
 
-  return end_result;
+    return result;
 }
 
-QString pythonSearchReg(const REG_SEARCH_TYPE type) {
-  HKEY key_handle1;
-  long res = 0;
+QString pythonSearchReg(const REG_SEARCH_TYPE type)
+{
+    HKEY hkRoot;
+    if (type == USER)
+        hkRoot = HKEY_CURRENT_USER;
+    else
+        hkRoot = HKEY_LOCAL_MACHINE;
+
+    REGSAM samDesired = KEY_READ;
+    if (type == SYSTEM_32BIT)
+        samDesired |= KEY_WOW64_32KEY;
+    else if (type == SYSTEM_64BIT)
+        samDesired |= KEY_WOW64_64KEY;
+
+    QString path;
+    LONG res = 0;
+    HKEY hkPythonCore;
+    res = ::RegOpenKeyExW(hkRoot, L"SOFTWARE\\Python\\PythonCore", 0, samDesired, &hkPythonCore);
```

Comment:
We open ```hkPythonCore``` here. We shouldn't close predefined handles like HKEY_CURRENT_USER, HKEY_LOCAL_MACHINE etc.